optuna/storages/_redis.py

```diff
@@ -691,12 +691,7 @@ def get_all_trials(
         states: Optional[Container[TrialState]] = None,
     ) -> List[FrozenTrial]:
 
-        trials = self._get_trials(study_id, states, set())
-
-        if deepcopy:
-            return copy.deepcopy(trials)
-        else:
-            return trials
+        return self._get_trials(study_id, states, set())
```

Comment:
Could you add a comment on why `deepcopy` is not necessary?

Reply:
Added. [581f448](https://github.com/optuna/optuna/pull/3672/commits/581f448ab53587c573b105288c1994b072f6d683)